app/interactors/file_attachments/update_interactor.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class FileAttachments::UpdateInteractor
+  include Interactor
+
+  delegate :params,
+           :user,
+           :edition,
+           :file_attachment_revision,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      find_and_update_file_attachment
+
+      update_edition
+      update_preview
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+  end
+
+  def find_and_update_file_attachment
+    current_attachment_revision = edition.file_attachment_revisions
+                                         .find_by!(file_attachment_id: params[:file_attachment_id])
+    attachment_params = params.require(:file_attachment).permit(:title, :file)
```

Comment:
Would we only need to permit the `title` at this point as we are not concerned about updating the `file` in this pr?

Reply:
Yes, that's one I've missed out removing - good catch.

Reply:
cool